setup/symfony_server.rst

```diff
@@ -0,0 +1,271 @@
+
+Different PHP Settings Per Project
+----------------------------------
+
+Selecting a Different PHP Version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have multiple PHP versions installed in your computer, you can tell
+Symfony which one to use creating a file called ``.php-version`` at the project
+root dir:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+
+    # use a specific PHP version
+    $ echo "7.2" > .php-version
+
+    # use any PHP 7.x version available
+    $ echo "7" > .php-version
+
+This other command is useful if you don't remember all the PHP versions
+installed on your computer:
+
+.. code-block:: terminal
+
+    $ symfony local:php:list
+
+Overriding PHP Config Options Per Project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can change the value of any PHP runtime config option per project by creating a
+file called ``php.ini`` at the project root dir. Add only the options you want
+to override:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+
+    # this project only overrides the default PHP timezone
+    $ cat php.ini
+    [Date]
+    date.timezone = Asia/Tokyo
+
+Running Commands with Different PHP Versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When running different PHP versions it's useful to use the main ``symfony``
+command as a wrapper of the ``php`` command. This allows you to always select the most
+appropriate PHP version according to the project which is running the commands:
+
+.. code-block:: terminal
+
+    # runs the command with the default PHP version
+    $ php -r "..."
+
+    # runs the command with the PHP version selected by the project
+    # (or the default PHP version if the project didn't select one)
+    $ symfony php -r "..."
+
+If you are using this wrapper frequently, consider aliasing the ``php`` command
+to it:
+
+.. code-block:: terminal
+
+    $ cd ~/.symfony/bin
+    $ cp symfony php
+    # now you can run "php ..." and the "symfony" command will be executed instead
+
+Local Domain Names
+------------------
+
+By default, projects are accessible at some random port of the ``127.0.0.1``
+local IP. However, sometimes it is preferable to associate a domain name to them:
+
+* It's more convenient when you work continuously on the same project because
+  port numbers can change but domains don't;
+* The behavior of some apps depend on their domains/subdomains;
+* To have stable endpoints, such as the local redirection URL of Oauth2.
+
+Setting up the Local Proxy
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Local domains are possible thanks to a local proxy provided by the Symfony
+server. First, start the proxy:
+
+.. code-block:: terminal
+
+    $ symfony proxy:start
+
+If this is the first time you run the proxy, you must follow these additional steps:
+
+* Open the **network configuration** of your operating system;
+* Find the **proxy settings** and select the **"Automatic Proxy Configuration"**;
+* Set the following URL as its value: ``https://127.0.0.1:7080/proxy.pac``
```

Comment:
I would add the image showing the window for the 3 operating systems (I can tweak the one I have in my slides for proper integration here).

Reply:
good idea indeed

Reply:
Good idea! Sadly the Windows screenshot in your presentation is in French. We'll need it in English.